src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringInterpolation.cs

```diff
@@ -130,44 +130,30 @@ private BoundExpression RewriteInterpolatedStringConversion(BoundConversion conv
                         : _factory.LogicalAnd(currentExpression, actualCall);
                 }
 
-                resultExpressions.Clear();
-
                 Debug.Assert(currentExpression != null);
 
-                var sequence = _factory.Sequence(
-                    appendShouldProceedLocal is not null
-                        ? ImmutableArray.Create(appendShouldProceedLocal.LocalSymbol)
-                        : ImmutableArray<LocalSymbol>.Empty,
-                    ImmutableArray.Create<BoundExpression>(handlerConstructionAssignment),
-                    currentExpression);
-
-                resultExpressions.Add(sequence);
+                resultExpressions.Clear();
+                resultExpressions.Add(handlerConstructionAssignment);
+                resultExpressions.Add(currentExpression);
             }
             else if (appendShouldProceedLocal is not null && resultExpressions.Count > 0)
             {
-                // appendCalls Sequence ending in true
-                var appendCallsSequence = _factory.Sequence(ImmutableArray<LocalSymbol>.Empty, resultExpressions.ToImmutableAndClear(), _factory.Literal(value: true));
+                // appendCalls as expressionStatements
+                var appendCallsStatements = resultExpressions.SelectAsArray(static (appendCall, @this) => (BoundStatement)@this._factory.ExpressionStatement(appendCall), this);
+                resultExpressions.Free();
 
-                resultExpressions.Add(handlerConstructionAssignment);
+                // if (appendShouldProceedLocal) { appendCallsStatements }
+                var resultIf = _factory.If(appendShouldProceedLocal, _factory.Block(appendCallsStatements));
 
-                // appendShouldProceedLocal && sequence
-                var appendAnd = _factory.LogicalAnd(appendShouldProceedLocal, appendCallsSequence);
-                var result = _factory.Sequence(ImmutableArray.Create(appendShouldProceedLocal.LocalSymbol), resultExpressions.ToImmutableAndClear(), appendAnd);
-
-                resultExpressions.Add(result);
-            }
-            else if (appendShouldProceedLocal is not null)
-            {
-                // Odd case of no append calls, but with an out param. We don't need to generate any jumps checking the local because there's
-                // nothing to short circuit and avoid, but we do need a sequence to hold the lifetime of the local
-                resultExpressions.Add(_factory.Sequence(ImmutableArray.Create(appendShouldProceedLocal.LocalSymbol), ImmutableArray<BoundExpression>.Empty, handlerConstructionAssignment));
+                _needsSpilling = true;
+                return new InterpolationHandlerResult(ImmutableArray.Create(_factory.ExpressionStatement(handlerConstructionAssignment), resultIf), builderTemp, appendShouldProceedLocal.LocalSymbol, _factory);
```

Comment:
>ImmutableArray.Create [](http://example.com/codeflow?start=54&length=21)

Do we ever pass more than one statement? Consider avoiding array allocation otherwise.